### PR TITLE
Changed add joints to use_joint_state_publisher | added joint state p…

### PR DESCRIPTION
**Pull request by delihus**
### Description

- to vizualize components with manipulators I added joint_state_publisher_gui,
- removed `add_wheel_joints`,
- added `use_joint_state_publiher`, `use_joint_state_publiher_gui`.

### Requirements

- [x] Code style guidelines followed
- [x] Documentation updated

### Tests 🧪

- [ ] Robot
- [x] Container
- [ ] Simulation

## Summary by CodeRabbit

- **New Features**
	- Added new launch arguments for simulation: `use_joint_state_publisher` and `use_joint_state_publisher_gui`.
	- Introduced a new launch configuration for visualizing a robot in RViz.
	- New configuration guide for setting up and visualizing a robot model using URDF.

- **Bug Fixes**
	- Updated default values for initial robot position parameters.

- **Documentation**
	- Enhanced README with clarifications on launch arguments and usage for both real and simulated environments.
	- Added detailed instructions for configuring and visualizing robot models.

- **Chores**
	- Added execution dependency on `joint_state_publisher_gui` in the package configuration.
	- Removed obsolete launch arguments related to wheel joints across multiple launch files. 

- **Configuration**
	- Introduced a new RViz configuration for improved visualization of the robot model.
	- Added example configuration for defining robot components in `components.yaml`.

### CODE REVIEW

File: husarion_ugv_gazebo/launch/simulate_robot.launch.py [hunk 1]
```diff
@@ -107,7 +107,6 @@ def generate_launch_description():
             )
         ),
         launch_arguments={
-            "add_wheel_joints": "False",
```
Comment (rafal-gorecki) [inline husarion_ugv_gazebo/launch/simulate_robot.launch.py:109]: This should not be removed.

File: lynx_description/launch/load_urdf.launch.py [hunk 2]
```diff
@@ -91,6 +82,22 @@ def generate_launch_description():
         description="Add namespace to all launched nodes.",
     )
 
+    use_joint_state_publisher = LaunchConfiguration("use_joint_state_publisher")
+    declare_use_joint_state_publisher_arg = DeclareLaunchArgument(
+        "use_joint_state_publisher",
+        default_value="False",
+        description="Flag enabling joint_state_publisher to publish information about joints positions. Should be false when there is a controller that sends this information.",
+        choices=["True", "true", "False", "false"],
+    )
```
Comment (rafal-gorecki) [inline lynx_description/launch/load_urdf.launch.py]: I someone just would like test robot_state_publisher and use this single launch file the wheels will not display correctly in rviz without `joint_state_publisher` node. 
```suggestion
    use_joint_state_publisher = LaunchConfiguration("use_joint_state_publisher", default="True")
```

**Review by rafal-gorecki — CHANGES_REQUESTED** [review]
1. For consistency and clarity I suggest: Make a general launch file called visualization.launch.py, which will contain auxiliary tools such as rviz rqt.
2. In fact, use_joint_state_publisher should never be changed by the user. So you can never declare it and use only LaunchConfiguration with default.
3. When you come back, the panther and lynx descriptions packages will most likely be merged. I suggest doing a PR after the merge.

File: husarion_ugv_gazebo/launch/simulate_robot.launch.py (hunk 1 above)
Comment (delihus) [reply husarion_ugv_gazebo/launch/simulate_robot.launch.py:109]: This parameter does not appear any more.

**Comment by delihus** [comment]
Waiting on https://github.com/husarion/panther_ros/pull/456

**Review by rafal-gorecki — APPROVED** [review]

**Pull request merged**